packages/legend-query/src/stores/QueryBuilderDateTimeHelper.ts

```diff
@@ -0,0 +1,587 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type Enum,
+  type PureModel,
+  EnumValueExplicitReference,
+  EnumValueInstanceValue,
+  GenericType,
+  GenericTypeExplicitReference,
+  matchFunctionName,
+  PrimitiveInstanceValue,
+  PRIMITIVE_TYPE,
+  SimpleFunctionExpression,
+  TYPICAL_MULTIPLICITY_TYPE,
+} from '@finos/legend-graph';
+import {
+  guaranteeNonNullable,
+  UnsupportedOperationError,
+} from '@finos/legend-shared';
+import {
+  DAY_OF_WEEK,
+  DURATION_UNIT,
+  SUPPORTED_FUNCTIONS,
+} from '../QueryBuilder_Const';
+import { buildPrimitiveInstanceValue } from './QueryBuilderOperatorsHelper';
+
+/**
```

Comment:
Let's put the comment directly above each attribute.

Also, let's restructure this slightly:

```ts
export type CustomDateOption = {
  label: string;
  value: string;
  unit?: string | undefined;
  adjustment?: {
    durationRange: string;
    adjustment: string;
    startDate: string;
  } | undefined;
};
```

Reply:
Also, why is `durationRange: string`? And we probably should rename `durationRange` to `duration`

Reply:
we need to discuss about `startDate`, it's slightly confusing to me